src/index.ts

```diff
@@ -0,0 +1,29 @@
+/**
+ * Initialization.
+ */
+import {Router} from './server';
+export const server = new Router();
+
+import {databases} from './database';
+
+console.log(`Server listening on ${server.port}`);
+
+process.on('uncaughtException', (err: Error) => {
+	Router.crashlog(err, 'The main process');
+});
+
+process.on('unhandledRejection', (err: Error) => {
+	Router.crashlog(err, 'A main process promise');
+});
+
+// graceful shutdown.
+process.on('SIGINT', () => {
+	void server.close().then(() => {
+		// we are no longer accepting requests and all requests have been handled.
+		// now it's safe to close DBs
+		for (const database of databases) {
+			database.close();
+		}
+		process.exit(0);
```

Comment:
It isn't necessary to explicitly exit the process here, and in fact by doing so you will hide graceful shutdown bugs. The process will naturally terminate when you have closed all open refs